packages/flutter_tools/lib/src/resident_devtools_handler.dart

```diff
@@ -116,6 +112,10 @@ class FlutterResidentDevtoolsHandler implements ResidentDevtoolsHandler {
       // report their URLs yet. Do so now.
       _residentRunner.printDebuggerList(includeVmService: false);
     }
+
+    final List<FlutterDevice?> devicesWithExtension = await _devicesWithExtensions(flutterDevices);
```

Comment:
Simply moving this seems to cause a race condition in `flutter error messages include a DevTools link` in `test/integration.shard/overall_experience_test.dart`. The test relies on the DevTools message to trigger a hot reload, which causes an overflow error that includes the DevTools URI. Unfortunately, the DevTools URI can only be output after the service extensions here have finished executing. Unless we want to put a delay in the test logic, we'll need to handle the `--start-paused` case separately here.

Reply:
@christopherfujino can you take a look at the updated changes when you have a chance?